### PR TITLE
Handle transliteration errors separately from other errors in Simple Forms

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR makes the simple forms controller specify if an error arises because of untransliteratable characters.

## Related issue(s)
https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/864

## Testing done

New test written but it's mysteriously failing.

## Screenshots
Instead of the scrubbed error, this is what we send to the FE:
Screenshot 2024-04-03 at 10 16 12 AM

### CODE REVIEW

**Review by pennja — APPROVED** [review]
I had one comment/possible suggestion but this otherwise looks good. Great work! 👍

File: modules/simple_forms_api/app/controllers/simple_forms_api/v1/uploads_controller.rb [hunk 1]
```diff
@@ -50,6 +50,8 @@ def submit
         else
           submit_form_to_central_mail
         end
+      rescue Prawn::Errors::IncompatibleStringEncoding
+        raise
```
[the remark this answers is not in the document]
Comment (Thrillberg) [reply modules/simple_forms_api/app/controllers/simple_forms_api/v1/uploads_controller.rb:54]: In doing this I learned that this raises the error that it rescued, exactly as it was, so it actually is descriptive (see screenshot for the full string that gets sent back to the FE).

[the remark this answers is not in the document]
Comment (pennja) [reply modules/simple_forms_api/app/controllers/simple_forms_api/v1/uploads_controller.rb:54]: Perfect, thanks!

**Review by pennja — APPROVED** [review]
LGTM!

**Review by pennja — DISMISSED** [review]
LGTM!

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**